Userland/Libraries/LibJS/Runtime/RegExpPrototype.cpp

```diff
@@ -240,7 +240,12 @@ static Value regexp_builtin_exec(GlobalObject& global_object, RegExpObject& rege
         }
 
         regex.start_offset = last_index;
-        result = regex.match(string);
+        // FIXME: JavaScript strings are UTF-16, update this if the backing storage
+        //        encoding changes for spec compliance reasons.
+        if (regex.options().has_flag_set(ECMAScriptFlags::Unicode))
```

Comment:
We have this set as a var above:
```suggestion
        if (unicode)
```

Reply:
Ah nice, I didn't notice :P